roles/openshift_sanitize_inventory/action_plugins/check_deprecated.py

```diff
@@ -0,0 +1,49 @@
+"""
+Ansible action plugin to check for usage of deprecated variables in Openshift Ansible inventory files.
+"""
+import sys
+import os
+import base64
+import traceback
+import errno
+import json
+from collections import defaultdict
+
+from ansible.plugins.action import ActionBase
+from ansible.module_utils.six import string_types
+
+try:
+    from __main__ import display
+except ImportError:
+    # pylint: disable=ungrouped-imports; this is the standard way how to import
+    # the default display object in Ansible action plugins.
+    from ansible.utils.display import Display
+    display = Display()
+
+class ActionModule(ActionBase):
+    """Action plugin to execute deprecated var checks."""
+
+    def run(self, tmp=None, task_vars=None):
+        result = super(ActionModule, self).run(tmp, task_vars)
+
+        deprecation_header = "[DEPRECATION WARNING]: The following are deprecated variables and will be no longer be used in the next minor release. Please update your inventory accordingly."
+
+        facts = dep_var_list = self._task.args.get('facts', '[]')
+        dep_var_list = self._task.args.get('vars', '')
+        dep_header = self._task.args.get('header', 'deprecation_header')
+
+        deprecation_message = dep_header
+        is_changed = False
+
+        for param in dep_var_list:
+            if param in facts:
+                deprecation_message = deprecation_message + "\n\t" + param
+                if not is_changed:
```

Comment:
This conditional seems unnecessary if we're breaking the loop anyway.  No need to check.